File: JAVA/K distance from root.java
class Tree
{
    void find(Node root,ArrayList<Integer> list,int k){
        if(root == null)return;
        if(k == 0)list.add(root.data);
        find(root.left,list,k-1);
        find(root.right,list,k-1);
    }
     ArrayList<Integer> Kdistance(Node root, int k)
     {
          ArrayList<Integer> list = new ArrayList<>();
          find(root,list,k);
          return list;
     }
}

File: C++/K distance from root.cpp
class Solution
{
    private: 
    void solve(Node* root, vector<int>&ans, int k, int count){
        if(!root) return;
        if(count == k){
            ans.push_back(root -> data);
        }
        solve(root->left, ans, k, count+1);
        solve(root->right, ans, k, count+1);
    }
    public:
    vector<int> Kdistance(struct Node *root, int k)
    {
      vector<int> ans;
      if(k == 0){
          ans.push_back(root -> data);
          return ans;
      }
      solve(root -> left, ans, k, 1);
      solve(root -> right, ans, k, 1);
      return ans;
    }
};
